pyccl/tests/test_ept_power.py

```diff
@@ -0,0 +1,260 @@
+                                       extra_params={'pad_factor': 2})
+    assert len(c.k_s) == 40
+
+
+@pytest.mark.parametrize('options', [['TG', 'TG', False, False],
+                                     ['TG', 'TG', False, True],
+                                     ['TG', 'TI', False, False],
+                                     ['TG', 'TM', False, False],
+                                     ['TI', 'TG', False, False],
+                                     ['TI', 'TI', False, False],
+                                     ['TI', 'TI', True, False],
+                                     ['TI', 'TM', False, False],
+                                     ['TM', 'TG', False, False],
+                                     ['TM', 'TI', False, False],
+                                     ['TM', 'TM', False, False]])
+def test_ept_get_pk2d_smoke(options):
+    if options[0] == options[1]:
+        t2 = None
+    else:
+        t2 = TRS[options[1]]
+    ptc = ccl.nl_pt.EulerianPTCalculator(
+        with_NC=True, with_IA=True, with_matter_1loop=True,
+        extra_params={'sub_lowk': options[3]},
+        cosmo=COSMO)
+    pk = ptc.get_pk2d_biased(TRS[options[0]],
+                             tracer2=t2,
+                             return_ia_bb=options[2])
+    assert isinstance(pk, ccl.Pk2D)
+
+
+def test_ept_pk2d_bb_smoke():
+    pee = PTC.get_pk2d_biased(TRS['TI'])
+    pbb = PTC.get_pk2d_biased(TRS['TI'], return_ia_bb=True)
+    assert pee.eval(0.1, 0.9, COSMO) != pbb.eval(0.1, 0.9, COSMO)
+
+
+@pytest.mark.parametrize('nl', ['nonlinear', 'linear', 'pt'])
+def test_ept_get_pk2d_nl(nl):
+    ptc = ccl.nl_pt.EulerianPTCalculator(
+        with_NC=True, with_IA=True, with_matter_1loop=True,
+        b1_pk_kind=nl, bk2_pk_kind=nl, cosmo=COSMO)
+    pk = ptc.get_pk2d_biased(TRS['TG'])
+    assert isinstance(pk, ccl.Pk2D)
+
+
+@pytest.mark.parametrize('typ_nlin,typ_nloc', [('linear', 'nonlinear'),
+                                               ('nonlinear', 'linear'),
+                                               ('nonlinear', 'pt'),
+                                               ('linear', 'pt')])
+def test_ept_k2pk_types(typ_nlin, typ_nloc):
+    tg = ccl.nl_pt.PTNumberCountsTracer(1., 0., 0., bk2=1.)
+    tm = ccl.nl_pt.PTNumberCountsTracer(1., 0., 0.)
+    ptc1 = ccl.nl_pt.EulerianPTCalculator(
+        with_NC=True, with_IA=True, with_matter_1loop=True,
+        b1_pk_kind=typ_nlin, bk2_pk_kind=typ_nloc,
+        cosmo=COSMO)
+    ptc2 = ccl.nl_pt.EulerianPTCalculator(
+        with_NC=True, with_IA=True, with_matter_1loop=True,
+        b1_pk_kind=typ_nloc, cosmo=COSMO)
+    pkmm = ptc1.get_pk2d_biased(tm, tracer2=tm)
+    pkmm2 = ptc2.get_pk2d_biased(tm, tracer2=tm)
+    pkgg = ptc1.get_pk2d_biased(tg, tracer2=tg)
+    ks = np.geomspace(1E-3, 1E1, 128)
+    p1 = pkgg.eval(ks, 1., COSMO)
+    p2 = pkmm.eval(ks, 1., COSMO)+ks**2*pkmm2.eval(ks, 1., COSMO)
+    assert np.all(np.fabs(p1/p2-1) < 1E-4)
```

Comment:
use `np.allclose`

Reply:
Why did you resolve all of these?

Reply:
Because they weren't worth the effort. Now with allclose so we can move on.

Reply:
Right, just keep that in mind in code you write/review hereafter. allclose is much easier to read, looking down the line